Duplicati/UnitTest/GeneralBlackBoxTesting.cs

```diff
@@ -36,13 +37,30 @@ protected IEnumerable<string> TestFolders
         {
             get
             {
-                var rx = new System.Text.RegularExpressions.Regex("r(?<number>\\d+)");
-                return 
-                    from n in Directory.EnumerateDirectories(SOURCE_FOLDERS)
-                    let m = rx.Match(n)
+                var rx = new Regex(@"r(?<number>\d+)");
+
+                try
+                {
+                    var folders = from n in Directory.EnumerateDirectories(SOURCE_FOLDERS)
+                        let m = rx.Match(n)
                         where m.Success
-                    orderby int.Parse(m.Groups["number"].Value)
-                    select n;
+                        orderby int.Parse(m.Groups["number"].Value)
+                        select n;
+
+                    var testFolders = folders as string[] ?? folders.ToArray();
+
+                    if (!testFolders.Any())
+                    {
+                        BasicSetupHelper.ProgressWriteLine($"Missing data in '{SOURCE_FOLDERS}' with required minimum of 1 sub-folder with the naming convention of '{rx}'");
+                    }
+
+                    return testFolders;
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    BasicSetupHelper.ProgressWriteLine($"Missing data directory '{SOURCE_FOLDERS}'");
+                    return null;
+                }

```

Comment:
These changes seem unrelated to the nuget and framework update.  Were these changes intended for a different pull request?

Reply:
Yes.. it can be removed from this PR.

Reply:
This PR is a bit messy. I'm thinking we just close it and if agreed I'll commit the 4.6.2 to a new PR.